planner/core/find_best_task.go

```diff
@@ -528,7 +528,7 @@ func (ds *DataSource) skylinePruning(prop *property.PhysicalProperty) []*candida
 			continue
 		}
 		// if we already know the range of the scan is empty, just return a TableDual
-		if len(path.Ranges) == 0 && !ds.ctx.GetSessionVars().StmtCtx.UseCache {
```

Comment:
There's the same logic at https://github.com/pingcap/tidb/blob/5c4e8dd4e46bd7c68bc97f917472159225dc9d61/planner/core/find_best_task.go#L688
I think we should do the same change there.

Reply:
We can use `OptimDependOnMutableConst` to replace `UseCache` check in other places as well, e.g, `refineArgs()`, but these changes are more like a refactor, not a bug fix, so I prefer not to do it in this PR.

Reply:
OK. Got it.
Sometimes really hard to understand the entire logic correctly.